packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts

```diff
@@ -0,0 +1,47 @@
+import { arrayFrom } from '@datadog/browser-core'
+
+export function getSelectorFromElement(targetElement: Element): string {
+  const targetElementSelector = []
+  let element: Element | null = targetElement
+
+  while (element && element.nodeName !== 'HTML') {
+    const idSelector = getIDSelector(element)
+    if (idSelector) {
+      targetElementSelector.unshift(idSelector)
+      break
+    }
+
+    targetElementSelector.unshift(getClassSelector(element) || getPositionSelector(element))
+
+    element = element.parentElement
+  }
+
+  return targetElementSelector.join('>')
+}
+
+function getIDSelector(element: Element): string | undefined {
+  const isUnique = element.id && element.ownerDocument.body.querySelectorAll(`#${element.id}`).length === 1
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏You should escape id and class names when using them as query selectors, see [CSS.escape](https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape)